compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1125,7 +1125,7 @@ rustc_queries! {
 
     query codegen_fulfill_obligation(
         key: (ty::ParamEnv<'tcx>, ty::PolyTraitRef<'tcx>)
-    ) -> Result<&'tcx ImplSource<'tcx, ()>, ErrorGuaranteed> {
+    ) -> Result<&'tcx ImplSource<'tcx, ()>, traits::CodegenObligationError> {
```

Comment:
Please pull this change out into a separate PR, it potentially fixes a lot of existing ICEs